students/paul_c/session04/mailroom.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/local/bin/python3
+
+import sys
+from datetime import datetime
+
+donors = {
+    "Anne Ant": [1.00],
+    "Bonnie Bug": [20.00, 10.00],
+    "Chuck Cat": [2.00, 2.00, 2.00],
+    "Donna Dog": [5000.00, 2500.00, 1.50],
+    "Edna Ent": [.30, .50, .10]
+}
+
+email = {
+    "greeting": "\nHello {}\n\n",
+    "body": "We would like to thank you for your generous donation of ${}.\n\n",
+    "closing": "Best Regards,\n",
+    "signature": "The Foundation\n\n"
+}
+
+
+def send_all():
+    """
+    Function to send a thank you letter to all donors.
+    This function sums total donations and writes letters to timestamped files.
+    """
+    for key, value in donors.items():
+       # I could not figure out how to get the timestamp to work with the filename for open(). I kept getting a file does not exist error.
+       # timestamp = datetime.now().strftime('%D')
+       # filename = "{}.txt.".format(key) + str(timestamp)
+        filename = "{}.txt".format(key)
```

Comment:
```
filename = f"{time.strftime("%b %d %Y %H:%M:%S")}-{key}.txt"
```

Reply:
Oh, this was a filesystem error, not a code error. Thanks for the tip!